### PR TITLE
External sso

### PR DESCRIPTION
**Pull request by akoserwal**

### CODE REVIEW

File: get_access_token.sh [hunk 1]
```diff
@@ -12,13 +12,27 @@ else
...
+
+            if [ ${?} -ne 0 ]; then
+                echo ${RHOAS_WHOAMI} > /dev/stderr
+                exit 1
+            fi
+
+            ACCESS_TOKEN=$(cat ${HOME}/.config/rhoas/config.json | jq -r '.access_token')
+            printf "%s\n" ${ACCESS_TOKEN}
+            exit 0
+        fi
```
Comment (MikeEdgar) [inline get_access_token.sh:34]: Currently no other way to obtain the "owner" token from sso.qa.redhat.com. We'll need a way to accomplish this without user interaction.

File: operators/install-kas-fleetshard.sh [hunk 2]
```diff
@@ -20,4 +20,31 @@ ${KUBECTL} wait --timeout=90s \
     deployment/kas-fleetshard-operator \
     --namespace=${NAMESPACE}
 
+if [ -n "${SSO_TRUSTED_CA:-}" ] ; then
```
Comment (b1zzu) [inline operators/install-kas-fleetshard.sh:23]: If not explicitly exported in the .env file the `SSO_TRUSTED_CA` is not "available" here, I think it should be explicitly exported in the kas-installer.sh

**Comment by k-wall** [comment]
@MikeEdgar @akoserwal where are we on this PR now?  Is Mike's comment above resolvable?  I know the work of @agullon is blocked at the minute.

**Comment by akoserwal** [comment]
> @MikeEdgar @akoserwal where are we on this PR now? Is Mike's comment above resolvable? I know the work of @agullon is blocked at the minute.

@k-wall Alejandra was able to successfully validate the test scenario. but the automation part is blocked due to the feature waiting from the sso team.  I'll check with them if we can have a workaround for now.

**Comment by MikeEdgar** [comment]
The only thing blocking this PR now a required change to kas-fleet-manager to map the clientId as the ManagedKafka owner rather than the preferred_username when the creator of the cluster is a service account.

**Comment by k-wall** [comment]
> The only thing blocking this PR now a required change to kas-fleet-manager to map the clientId as the ManagedKafka owner rather than the preferred_username when the creator of the cluster is a service account.

Is there a JIRA or issue tracking this change?

**Comment by StevenTobin** [comment]
@MikeEdgar @k-wall @akoserwal could I get an update on the status of this PR? We have some fault tests that not having this in place is affecting.

**Comment by Rajagopalan-Ranganathan** [comment]
@MikeEdgar @akoserwal  - We need to update the readme for sso.qa env, and call out this would only work with red hat VPN.

**Comment by MikeEdgar** [comment]
@Rajagopalan-Ranganathan I'll update this PR and rebase on the changes to `main`. I think we need to come up with an approach that will work without VPN.

**Comment by MikeEdgar** [comment]
Replaced by #173

**Pull request closed without merging**